src/main/java/org/killbill/billing/plugin/vertex/TaxItemDetails.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020-2024 Equinix, Inc
+ * Copyright 2020-2024 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.plugin.vertex;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.killbill.billing.invoice.usage.details.UsageConsumableInArrearAggregate;
+import org.killbill.billing.invoice.usage.details.UsageConsumableInArrearTierUnitAggregate;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Wrapper to add taxRate details into itemDetails
+ */
+public class TaxItemDetails {
+
+    private final List<UsageConsumableInArrearTierUnitAggregate> tierDetails;
+    private final BigDecimal amount;
+    private final Double taxRate;
+
+    public TaxItemDetails(@Nonnull final Double taxRate, @Nullable final UsageConsumableInArrearAggregate usageDetail) {
+        this(taxRate, usageDetail != null ? usageDetail.getTierDetails() : null, usageDetail != null ? usageDetail.getAmount() : null);
+    }
+
+    @JsonCreator
+    public TaxItemDetails(@JsonProperty("taxRate") final Double taxRate,
+                          @JsonProperty("tierDetails") List<UsageConsumableInArrearTierUnitAggregate> tierDetails,
+                          @JsonProperty("amount") BigDecimal amount) {
+        this.tierDetails = tierDetails;
+        this.amount = amount;
+        this.taxRate = taxRate;
+    }
```

Comment:
what if Killbill adds more details than just tiers and amount. itemDetails are kind of blackbox to us (as it could contain any information not just tiers/amount) but we know its a Json. Can't we do something like below without even having to use object TaxItemDetails for the same.
```

                ObjectMapper mapper = new ObjectMapper();
                JsonNode actualObj = mapper. readTree(itemDetails);
                ((ObjectNode) actualObj).put("taxRate", 1.0);
```

Reply:
I think that's great suggestion, I didn't like the way how vertex plugin became exposed to invoice module implementation details of KB with version fetch in pom.xml as well

Reply:
fixed in 028ba4ac90ece5a10c37259feb7eb8f25f1eb9e1